### PR TITLE
[SDK-416] Support app level overrides for the pre-install data

### PR DESCRIPTION
**Pull request by code-crusher**
**Description:**
As per the requirement, we also need to ensure the app level data set overrides the value in the system prop file. 

**Testing:**
- [X] Manual
- [X] Unit Test

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/Branch.java [hunk 1]
```diff
@@ -993,6 +993,9 @@ public Branch setPreinstallPartner(@NonNull String preInstallPartner) {
         return this;
     }
 
+     String getInstallMetaData(String key) {
```
Comment (apeterson-branch) [inline Branch-SDK/src/io/branch/referral/Branch.java]: Theoretically... you can get PrefHelper directly without having to go through Branch to do so.

`PrefHelper.getInstance(context)`

That way you don't have to add another method to Branch.java.

File: Branch-SDK/src/io/branch/referral/PrefHelper.java [hunk 2]
```diff
@@ -29,8 +29,8 @@
     /**
      * The base URL to use for all calls to the Branch API.
      */
-    static final String BRANCH_BASE_URL_V2 = "https://api2.branch.io/";
-    static final String BRANCH_BASE_URL_V1 = "https://api.branch.io/";
+    static final String BRANCH_BASE_URL_V2 = "https://release.api.beta.branch.io/";
```
Comment (apeterson-branch) [inline Branch-SDK/src/io/branch/referral/PrefHelper.java]: This can't be changed.

File: Branch-SDK/src/io/branch/referral/PrefHelper.java [hunk 3]
```diff
@@ -29,8 +29,8 @@
...
+    static final String BRANCH_BASE_URL_V1 = "https://release.api.beta.branch.io/";
```
Comment (apeterson-branch) [inline Branch-SDK/src/io/branch/referral/PrefHelper.java]: This can't be changed.

File: Branch-SDK/src/io/branch/referral/Branch.java (hunk 1 above)
Comment (code-crusher) [reply Branch-SDK/src/io/branch/referral/Branch.java]: 👌

File: Branch-SDK/src/io/branch/referral/PrefHelper.java (hunk 2 above)
Comment (code-crusher) [reply Branch-SDK/src/io/branch/referral/PrefHelper.java]: Test leak, fixed.

File: Branch-SDK/src/io/branch/referral/PrefHelper.java (hunk 3 above)
Comment (code-crusher) [reply Branch-SDK/src/io/branch/referral/PrefHelper.java]: Test leak, fixed.

**Comment by code-crusher** [comment]
@apeterson-branch 

Changes made:
1. Used `PrefHelper` instance.
2. Revert the API endpoints.
3. Added comments
4. Revert the additional method in `Branch.java`

**Pull request merged**